packages/ses/test/test-freeze-module-namespace.js

```diff
@@ -0,0 +1,22 @@
+// import '../index.js';
+// import './lockdown-safe.js';
+import test from 'ava';
+
+import * as namespace from './module-namespace-const-1.js';
+
+const { freeze, isFrozen, isExtensible, getOwnPropertyDescriptors } = Object;
+
+test('freeze module-namespace-const-1', t => {
+  t.is(isFrozen(namespace), false, 'not pre-frozen');
+  t.is(isExtensible(namespace), false, 'is pre non-extensible');
+  const preDescs = getOwnPropertyDescriptors(namespace);
+  t.is(preDescs.one.writable, true, 'export const is unfortunately writable');
```

Comment:
Yeah I've wondered why exports are writable. Doesn't make much sense since the exotic module namespace object will prevent Set.

Reply:
Actually thinking of it, I suppose it's because the value can change (for `let` exports), just not by assignment to the exotic object, but if it was defined as non-writable and non-configurable, some code could assume it'd be frozen and never change (invariant)

Reply:
But since it is `const` it cannot change once initialized.

If you access before it is initialized (which can only happen in a cycle, IIUC), then you encounter temporal dead zone (TDZ) and it throws. Transitioning from throwing to reporting a stable value does not violate the non-writable, non-configurable invariants. A frozen proxy can also throw when it wants to. The invariants only constrain what it does when it does not throw.

Thus, I would argue that the spec should be changed so the namespace property for each exported `const` binding be a non-writable, non-configurable data property, whose observably exotic behavior is only that it throws if accessed during TDZ.